### DISCUSSION TITLE
BalloonSat CubeSatSim

### DISCUSSION THREAD

**Discussion by GRBlake488** [alanbjohnston/CubeSatSim#256] [Ideas]
Greetings. I am modifying the AmSat / Alan johnston CubeSatSimulator to be used on a weather balloon. I am hoping to launch it Summer 2023.

I have updated the design in several ways:

1. I changed the ArduCam to the newer ArduCam with Auto-Focus for better viewing.
2. I have added an RTC module for time/date stamping and also for software clocking.
3. I have added a 2 watt 433 MHz Amplifier on a Perma-Proto Pi Hat board. This is going to switch in for 5 minutes once every time the CSS is in APRS mode for extended recption.
4. I am changing the software to step through each Mode every 30 minutes so that anyone will be able to receive some mode of communications from the Balloon.
5. I plan to add a GPS Receiver to the board stack.
6. Modify the STL files to make a better, larger frame.
7. I am going to change the three-battery holder to a single battery for flight and endurance.

I would like to know if anyone has changed the Code to add GPS data? This would be a big help. Also, which GPS Receiver did you use / recommend? 

Let me know.

73's

George B;lake
W6BDD

**Comment by Boovtech** [comment]
Hi George,
 
I like your idea and I'd love to see your revised flight-worthy cubesat sim.  And of course I'd like to know when you get it launched so I can listen in.

73
Virginia NV5F

**Comment by alanbjohnston** [comment]
Hi George,

Thanks for sharing your balloon payload plans - good luck with them!

I have played around with GPS code and also flown the CubeSatSim Lite twice as a balloon payload.

The first time, I used a GPS with a serial connection to talk to the STM32 on the STEM Payload board.  Here's the code: https://github.com/alanbjohnston/CubeSatSim/blob/bp/arduino/Payload_BME280_MPU6050_XS_GPS.ino

The code worked OK, but the GPS board cut off at 20,000 feet so it was mostly useless on a balloon.

The second time, I used the gpsd library to talk to a USB connected GPS dongle plugged into the Pi Zero.  The code that I used is in the bp-new branch: https://github.com/alanbjohnston/CubeSatSim/tree/bp-new

Unfortunately this setup was extremely unreliable.  It would take over 10 minutes for the GPS to get a fix, and sometimes it would never get a fix at all.  And only a complete power cycle (even a reset of the USB port didn't help) would give it another chance.  In the end, this GPS also failed in flight, probably due a power cycle then not getting a fix for the rest of the flight.

I did use a script (in the bp-new branch) that cycled through the modes.  I also wrote some code take and store photos every 30 seconds too.

Maybe some of this code will be of use to you, or maybe it will just help you avoid things that don't work.

You might also be interested in a new effort underway to redesign the CubeSatSIm to use a Raspberry Pi Pico as the processor instead of the Pi Zero.  This started because of the Pi Zero shortage, but it should also be a better design to fly, since a microcontroller like the Pico is more stable than a Linux SBC like the Pi Zero in adverse conditions.

You can see some of the code in the pico branch: https://github.com/alanbjohnston/CubeSatSim/tree/pico

You can also see some early versions of the Pico PCB here: https://github.com/alanbjohnston/CubeSatSim/tree/pico/hardware/pico

The board uses a SR_FRS_05W which is a 0.5 W FM transmitter for APRS and SSTV modes.  It uses a Si5351 clock generator for FSK and BPSK modes.

We are hoping to release a Beta of the hardware and software early in the new year.

Keep us posted with your balloon flight.  And after your flight, let me know if you would be interested in sharing it a short article in the AMSAT Journal.  There are plenty of others who have built CubeSatSims and are also interested in balloon payloads.

73,
Alan
KU2Y

**Comment by GRBlake488** [comment]
Hello Alan,

Thanks for the reply.

First, I have a question: I don't have the Current Sensors nor the Temp / Pressure Sensor installed yet because of the shortages. I am supposed to have them by February. When I try my modified software and I only operate from battery, when I push the button it resets the CPU instead of stepping through the options. I loaded the original onto an SD card and it does the same thing. If I power the PI direct through the USB Power port, it works fine. Does it need a full complement of electronics to operate correctly? I am seeing a transmitted signal (albeit very low) out of the RF port. So, it appears to work. Any thoughts?

Next, I bought a 2 Watt RF Amplifier module for 433 MHz. My plan is to burst the data signal once every thirty minutes for maybe 2 minutes by switching in the amplifier. That way, it can be tracked easier. It would be nice if we could transmit on APRS or a Ham band for that period. I also bought a 2.4 GHz Transceiver board I found online and I may use that to transmit 2.4 GHz, and also receive 2.4 GHz. 

I need to modify the code to allow stepping of the modes from 1 to 2 to 3, etc. That way, data will be sent out on all modes. I am also buildign a two-axis tracking system with 145 / 442 / 2400 MHz. The 2400 MHz is going to be for HamTV on the ISS (if they ever get it back up), but I can use it for the Transceiver function as well.

I bought the ArduCam with Auto-Focus and installed the software for that. It appears to work well. I also added 2.4 GHz WiFi so I can stream the video as I wish. It doesn't appear to affect the SSTV functoion.

I also added an RTC board. You should consider that as well.

Building in the GPS would be helpful. I understand it was built to be a Demo unit, but having these functionalities would be useful for projects like mine. It makes it more diverse. Or, perhaps another version?

I am changing the battery set out to one battery with higher capacity. I don't trust AA's and a battery holder for flight travel.

The STL frame is not good to mount those Solar cells. The openings are too big to mount them. You need a substrate to hold them and attach to the 3D frames.. I am redesigning them also as well as the increased size.

The STEM circuitry: With everything on the CSS I'm not sure what else you would want to demonstrate. There is plenty of functions to show.

For the balloon: My Sister is a Florist. She told me about this stuff called HI FLOT that they squirt into Latex balloons to keep them inflated longer. it is like FIX-A-FLAT only for balloons. It coats the inside of the membrane and reduces the leakage through the balloon. You inflate the balloon enough to allow it to get on all surfaces, then squirt it in, and roll the balloon around. In theory it should work on a weather ballon and keep it aloft longer.

I look forward to your reply.

73's.

George Blake

**Comment by GRBlake488** [comment]
BTW:

1. Where is the code found that sets the frequency for transmit?
2. Is there a way to output the raw data on a pin to use for other transmitters? Serial pins??

Thanks.

**Reply by alanbjohnston** [reply]
Hi George,

1. All the transmit code for the Pi Zero is in this Python code: https://github.com/alanbjohnston/CubeSatSim/blob/master/rpitx.py

if you search for 434.9 you will find all the places it is set. Note that you may need a different Low Pass Filter if you change it. 

2. sensor data can be shared on a serial connection. That is how the sensor data from the STEM Payload board is shared with the Pi Zero. The STM32 controller has a second serial port which you could use for this. The Pi could also be setup with another serial port. 

A better way to do it might be to just connect the I2C bus that the sensors are on to your other board. That way, it could talk to the sensors directly. 

Good luck and 73,
Alan

**Comment by GRBlake488** [comment]
Alan,
1. That is what I suspected when I looked at the code. I will give it a try before I change the filter. What is the maximum frequency you can transmit at? Or have you tried it? Of course, with increase of frequency comes loss in power.
2. I would rather have a buffered output vs. tapping the sensors directly. At least, that way, it is assumed to be formatted. Also, if there is a short it won't pull the sensors down.
3. Any thoughts on stepping the modes? It looks like that would also be addded in the same rpitx.py file?
73's
George Blake








    On Thursday, January 5, 2023, 01:05:53 AM CST, alanbjohnston ***@***.***> wrote:  
 
 


Hi George,
   
   - All the transmit code for the Pi Zero is in this Python code: https://github.com/alanbjohnston/CubeSatSim/blob/master/rpitx.py

if you search for 434.9 you will find all the places it is set. Note that you may need a different Low Pass Filter if you change it.
   
   - sensor data can be shared on a serial connection. That is how the sensor data from the STEM Payload board is shared with the Pi Zero. The STM32 controller has a second serial port which you could use for this. The Pi could also be setup with another serial port.

A better way to do it might be to just connect the I2C bus that the sensors are on to your other board. That way, it could talk to the sensors directly.

Good luck and 73,
Alan
